agent/consul/state/catalog_test.go

```diff
@@ -3636,6 +3635,449 @@ func TestStateStore_CheckConnectServiceNodes_Gateways(t *testing.T) {
 	assert.Equal(443, nodes[1].Service.Port)
 }
 
+func TestStateStore_CheckGatewayServiceNodes_Terminating(t *testing.T) {
+	assert := assert.New(t)
```

Comment:
Looks fine for this test but usually I like to stay away from shadowing the packages name. It would complicate things if there were subtests or retry blocks.